service/src/test/java/bio/terra/drshub/controllers/DrsHubApiControllerTest.java

```diff
@@ -144,6 +146,48 @@ void testCallsCorrectEndpointsWhenOnlyAccessUrlRequestedWithPassportsUsingFallba
 
     // need an extra verify because nothing in the mock cares that bearer token is set or not
     verify(drsApi).setBearerToken(TEST_BOND_SA_TOKEN);
+    // verify that the passport postAccessURL method was not called, since there is no passport
+    verify(drsApi, never()).postAccessURL(any(), any(), any());
+  }
+
+  @Test
+  void testFallbackWhenOnlyAccessUrlRequestedWithPassportsFails() throws Exception {
+    var accessId = "gs";
+    var cidProviderHost = getProviderHosts("passport");
+    var drsObject = drsObjectWithRandomId("gs");
+
+    var drsApi =
+        mockDrsApiAccessUrlWithToken(
+            cidProviderHost.dnsHost(), drsObject, accessId, TEST_ACCESS_URL);
+
+    when(drsApi.postAccessURL(
+            Map.of("passports", List.of(TEST_PASSPORT)), drsObject.getId(), accessId))
+        .thenThrow(new RestClientException("Failed to retrieve access url with passport"));
+
+    mockExternalcredsApi("ras", TEST_ACCESS_TOKEN, Optional.of(TEST_PASSPORT));
+
+    mockBondLinkAccessTokenApi(
+        cidProviderHost.drsProvider().getBondProvider().get(),
```

Comment:
This is just test code but using `get()` without `isPresent()` is unsafe; to make the unsafe-ness explicit I would use `orElseThrow()` instead of `get()`.
```suggestion
        cidProviderHost.drsProvider().getBondProvider().orElseThrow(),
```